packages/tao/src/commands/migrate.ts

```diff
@@ -431,45 +430,66 @@ async function generateMigrationsJsonAndUpdatePackageJson(
     to: { [p: string]: string };
   }
 ) {
-  logger.info(`Fetching meta data about packages.`);
-  logger.info(`It may take a few minutes.`);
-  const migrator = new Migrator({
-    versions: versions(root, opts.from),
-    fetch: createFetcher(logger),
-    from: opts.from,
-    to: opts.to
-  });
-  const { migrations, packageJson } = await migrator.updatePackageJson(
-    opts.targetPackage,
-    opts.targetVersion
-  );
-  updatePackageJson(root, packageJson);
+  try {
+    logger.info(`Fetching meta data about packages.`);
+    logger.info(`It may take a few minutes.`);
+    const migrator = new Migrator({
+      versions: versions(root, opts.from),
+      fetch: createFetcher(logger),
+      from: opts.from,
+      to: opts.to
+    });
+    const { migrations, packageJson } = await migrator.updatePackageJson(
+      opts.targetPackage,
+      opts.targetVersion
+    );
+    updatePackageJson(root, packageJson);
 
-  if (migrations.length > 0) {
-    createMigrationsFile(root, migrations);
+    if (migrations.length > 0) {
+      createMigrationsFile(root, migrations);
 
-    logger.info(`The migrate command has run successfully.`);
-    logger.info(`- package.json has been updated`);
-    logger.info(`- migrations.json has been generated`);
+      logger.info(`The migrate command has run successfully.`);
+      logger.info(`- package.json has been updated`);
+      logger.info(`- migrations.json has been generated`);
 
-    logger.info(`Next steps:`);
-    logger.info(
-      `- Make sure package.json changes make sense and then run 'npm install' or 'yarn'`
+      logger.info(`Next steps:`);
+      logger.info(
+        `- Make sure package.json changes make sense and then run 'npm install' or 'yarn'`
+      );
+      logger.info(`- Run 'nx migrate --run-migrations=migrations.json'`);
+    } else {
+      logger.info(`The migrate command has run successfully.`);
+      logger.info(`- package.json has been updated`);
+      logger.info(
+        `- there are no migrations to run, so migrations.json has not been created.`
+      );
+    }
+  } catch (e) {
+    const startVersion = versions(root, {})(opts.targetPackage);
+    logger.error(
+      `The migrate command failed. Try the following to migrate your workspace:`
     );
-    logger.info(`- Run 'nx migrate --run-migrations=migrations.json'`);
-  } else {
-    logger.info(`The migrate command has run successfully.`);
-    logger.info(`- package.json has been updated`);
-    logger.info(
-      `- there are no migrations to run, so migrations.json has not been created.`
+    logger.error(
+      `> npm install --save-dev ${opts.targetPackage}@${opts.targetVersion}`
     );
+    logger.error(
+      `> nx migrate ${opts.targetVersion} --from="@nrwl/workspace@${startVersion}"`
```

Comment:
So this would print something like:

```
nx migrate 9.0.0 --from"@nrwl/workspace@8.9.0
```

Shouldn't it print:

```sh
nx migrate @nrwl/workspace@9.0.0 --from"@nrwl/workspace@8.9.0
```

Also, shouldn't the package be a variable as well?

Reply:
you are right